### ISSUE TITLE
Is type casting of void* parameters necessary?

### ISSUE THREAD

**Issue by rabe-soft** [bluewaysw/pcgeos#179]
Void parameters of functions require explicit type casting to (void*). Example:
MemUnlockFixedOrMovable(void* ptr); requires MemUnlockFixedOrMovable((void*)&TSC_dupInfo);
In my opinion, this contradicts the idea that void* should be assignment compatible with all pointers.

**Comment by bluewaysw** [comment]
In the example given (as in the UICtrl SDK_C sample) the issue is that the original pointer is const and the target isn't. 

Do you have more examples having this issue @rabe-soft ?

We still may use a pragma to disable this specific warning.

**Comment by rabe-soft** [comment]
You are right, some times no type casting is reqired. The following code comes from my Uni Installer app.  HugeArrayLock() requires (void**), but HugeArrayUnlock() requires no type casting.

InstallEntry * IFLockMainArrayElement(word item) {
VMBlockHandle	mainArray;
InstallEntry 	*elemPtr;
word 		size;

   mainArray = IFGetMainArrayHandle();
   HugeArrayLock(global.docFile,mainArray,item, (void**)&elemPtr, &size);
   return	elemPtr;
   }
void	IFUnlockMainArray(InstallEntry 	*elemPtr ) {
	HugeArrayUnlock(elemPtr);
	}
